### PR TITLE
docs: Add alternative syntax for extract, trim and substring.

### PR DESCRIPTION
**Pull request by Omega359**
## Which issue does this PR close?

Related to #13139, discussed in #13036

## Rationale for this change

Complete documentation for functions with alternative syntax 

## What changes are included in this PR?

doc updates only

## Are these changes tested?

Yes.

## Are there any user-facing changes?

Documentation

### CODE REVIEW

File: datafusion/functions/src/string/btrim.rs [hunk 1]
```diff
@@ -124,6 +124,8 @@ fn get_btrim_doc() -> &'static Documentation {
 ```"#)
             .with_standard_argument("str", Some("String"))
             .with_argument("trim_str", "String expression to operate on. Can be a constant, column, or function, and any combination of operators. _Default is whitespace characters._")
+            .with_alternative_syntax("trim(BOTH trim_str FROM str)")
+            .with_alternative_syntax("trim(trim_str FROM str)")
```
Comment (findepi) [inline datafusion/functions/src/string/btrim.rs:128]: Does this deserve new SLT test cases?

**Review by findepi — APPROVED** [review]

File: datafusion/functions/src/string/btrim.rs (hunk 1 above)
Comment (Omega359) [reply datafusion/functions/src/string/btrim.rs:128]: I grabbed the syntax from an slt test :)

**Comment by jatin510** [comment]
lgtm 👍

**Review by jonathanc-n — APPROVED** [review]
This looks good to me. We're pretty close!

**Review by alamb — APPROVED** [review]
Thank you @Omega359 @findepi and @jonathanc-n

**Pull request merged**